Infrastructure.py

```diff
@@ -0,0 +1,152 @@
+                            current_order_id = col
+                            current_order = Order(col)
+                            current_pick_data = current_pick_data + [col]
+                        else:
+                            current_pick_data = current_pick_data + [col]
+                            previous_order_id = current_order_id
+                    else:
+                        current_pick_data = current_pick_data + [col]                        
+                colnum += 1
+            if (rownum > 0):
+                #populate the pick object with the obtained data
+                pick = Pick(current_pick_data)
+                current_order.picks = current_order.picks + [pick]
+            rownum += 1
+        #add the last order to the array
+        orders[current_order_id] = current_order
+    return orders
+
+class Order:
+    def __init__(self, order_id):
+        self.picks = []
+        self._order_id = order_id
+
+    #helper function
+    def numPicks(self):
+        return len(self.picks)
+
+class Batch:
+    def __init__(self):
+        self.picks = []
+
+
+class Warehouse:
+    #initialize the matrix which will be populated with the distances from csv
+    def __init__(self):
+        self.dist = {}
+    
+    def read_distances(self, data_file):
+        """Function which reads the csv of the distances between the nodes and
+           returns a dictionary of dictionaries with keys of the node id
+
+           Args:
+               data_file (string): name of the csv file.
+           
+           Returns:
+               dictionary: distances between the nodes.
+
+           Example:
+
+           >>> warehouse = Warehouse() #initialize warehouse object
+           >>> warehouse.read_distances("test.csv") #read the distances
+           >>> warehouse.dist["a"] #print all the distances from node "a"
+           {'b': '23', 'c': '4', 'a': '5'}
+           >>> warehouse.dist["a"]["b"] #get the distance between node "a" and "b"
+           '23'
+
+        """
+        #open the file
+        with open(data_file, 'r') as datafile:
+            reader = csv.reader(datafile, delimiter=';')
+            array_nodes = []
+            rownum = 0
+            #iterate through each element in the csv and avoid the headers
+            for row in reader:
+                colnum = 0
+                current_dictionary = {}
+                current_node = ""
+                for col in row:
+                    if rownum == 0:
+                        array_nodes = array_nodes + [col]
+                    else:
+                        if colnum == 0:
+                            current_node = col
+                        else:
+                            current_dictionary[array_nodes[colnum]] = col
+                    colnum += 1
+                self.dist[current_node] = current_dictionary
+                rownum += 1
+        return self.dist
+
+class Pick:
+    def __init__(self, data_row):
```

Comment:
Here I it would be nice a comment on what kind of data object this data_row is. Also for all the attributes you are initialising, what data types are they and what are they ment for.